src/frontend/src/app/assets/assets.component.ts

```diff
@@ -7,14 +7,35 @@ import { AssetsService, AlertService } from '../services/index';
   styleUrls: ['./assets.component.css']
 })
 export class AssetsComponent implements OnInit {
+  public limit: Number = 0;
+  public offset: Number = 0;
   assets = [];
   assetsReadingsData = [];
+  asset_code;
   constructor(private assetService: AssetsService, private alertService: AlertService) { }
 
   ngOnInit() {
     this.getAsset();
   }
 
+  public setLimit(limit: Number) {
+    this.asset_code = <HTMLSelectElement>document.getElementById('code');
+    this.limit = limit;
+    console.log('Limit: ', limit);
+    console.log('asset_code: ', this.asset_code.value);
+    this.getAssetReading(this.asset_code.value);
+  }
+
+  public setOffset(offset: Number) {
+    this.asset_code = <HTMLSelectElement>document.getElementById('code');
+    this.offset = offset;
+    console.log('offset:', offset);
+    console.log('asset_code: ', this.asset_code.value);
+    if (this.limit != null) {
+      this.getAssetReading(this.asset_code.value);
```

Comment:
asset_code is a class level variable so no need to pass asset_code to getAssetReading method as a parameter. Call directly into getAssetReading .

Reply:
method `getAssetReading` expecting a variable, so it is needed to pass asset_code to the method.